File: backend/app/schemas/worker.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum


class WorkerRole(str, Enum):
    POTONG = "Potong"
    JAHIT = "Jahit"
    FINISHING = "Finishing"
    MAGANG = "Magang"
    OTHER = "Other"


class WorkerStatus(str, Enum):
    WORKING = "Working"
    IDLE = "Idle"


class WorkerBase(BaseModel):
    name: str
    role: WorkerRole
    status: Optional[WorkerStatus] = WorkerStatus.IDLE

class WorkerUpdate(BaseModel):
    name: Optional[str] = None
    role: Optional[WorkerRole] = None
    status: Optional[WorkerStatus] = None

class Worker(WorkerBase):
    id: int
    date_joined: datetime
    is_deleted: bool = False

    class Config:
        from_attributes = True

# Performance
class DailyPerf(BaseModel):
    date: str
    count: int


class WorkerPerformance(BaseModel):
    worker_id: int
    worker_name: str
    performance_score: float
    total_finished: int
    daily: List[DailyPerf] = []


# Tasks
class WorkerTask(BaseModel):
    log_id: int
    order_item_id: int
    receipt_number: str
    customer_name: str
    garment_type: str
    status: str
    completed_at: str

File: frontend/shared/types/worker.ts
export const WORKER_ROLES = ['Potong', 'Jahit', 'Finishing', 'Magang', 'Other'] as const;
export type WorkerRole = typeof WORKER_ROLES[number];

export const WORKER_STATUSES = ['Working', 'Idle'] as const;
export type WorkerStatus = typeof WORKER_STATUSES[number];

// -- Worker --
export interface WorkerCreate {
  name: string;
  role: WorkerRole;
  status?: WorkerStatus;
}

export interface WorkerUpdate {
  name?: string | null;
  role?: WorkerRole | null;
  status?: WorkerStatus | null;
}

export interface Worker {
  id: number;
  name: string;
  role: WorkerRole;
  status?: WorkerStatus;
  date_joined: string; // ISO DateTime string
}

export interface DailyPerf {
  date: string;
  count: number;
}

export interface WorkerPerformance {
  worker_id: number;
  worker_name: string;
  performance_score: number;
  total_finished: number;
  daily: DailyPerf[];
}

export interface WorkerTask {
  log_id: number;
  order_item_id: number;
  receipt_number: string;
  customer_name: string;
  garment_type: string;
  status: string;
  completed_at: string;
}